src/lib/mina_lib/mina_lib.ml

```diff
@@ -903,20 +903,10 @@ let add_work t (work : Snark_work_lib.Result.Partitioned.Stable.Latest.t) =
   | Processed None ->
       `Ok
   | Processed (Some (stmts, priced_proof)) ->
-      let cb _ =
```

Comment:
This eagerly remove jobs completed from selector's reschedule pool. Since it no longer take cares of reschedule, but just filter out works already sent. We could clean it up lazily when new frontier arrives

Reply:
Same goes for other `remove_old_assignments` calls